src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/GeneralApplicationDirectionsService.java

```diff
@@ -0,0 +1,184 @@
+            GENERAL_APPLICATION_DIRECTIONS_NWYORKSHIRE_COURT,
+            GENERAL_APPLICATION_DIRECTIONS_HUMBER_COURT,
+            GENERAL_APPLICATION_DIRECTIONS_KENTSURREY_COURT,
+            GENERAL_APPLICATION_DIRECTIONS_NEWPORT_COURT,
+            GENERAL_APPLICATION_DIRECTIONS_SWANSEA_COURT,
+            GENERAL_APPLICATION_DIRECTIONS_ADDITIONAL_INFORMATION,
+            GENERAL_APPLICATION_DIRECTIONS_RECITALS,
+            GENERAL_APPLICATION_DIRECTIONS_JUDGE_TYPE,
+            GENERAL_APPLICATION_DIRECTIONS_JUDGE_NAME,
+            GENERAL_APPLICATION_DIRECTIONS_COURT_ORDER_DATE,
+            GENERAL_APPLICATION_DIRECTIONS_TEXT_FROM_JUDGE
+        ).forEach(generalApplicationDirectionCcdField -> caseData.remove(generalApplicationDirectionCcdField));
+    }
+
+    public void submitGeneralApplicationDirections(CaseDetails caseDetails, String authorisationToken) {
+        CaseDocument document = caseDetails.getData().get(GENERAL_APPLICATION_DIRECTIONS_HEARING_REQUIRED).equals(YES_VALUE)
+            ? prepareHearingRequiredNoticeDocument(caseDetails, authorisationToken)
+            : prepareGeneralApplicationDirectionsOrderDocument(caseDetails, authorisationToken);
+
+        printDocumentPackAndSendToApplicantAndRespondent(caseDetails, authorisationToken, document);
+        updateGeneralApplicationLatestDocument(caseDetails, document);
+        resetStateToGeneralApplicationPrestate(caseDetails);
+    }
+
+    private void printDocumentPackAndSendToApplicantAndRespondent(CaseDetails caseDetails, String authorisationToken, CaseDocument document) {
+        List<BulkPrintDocument> documents = asList(documentHelper.getCaseDocumentAsBulkPrintDocument(document));
+        bulkPrintService.printApplicantDocuments(caseDetails, authorisationToken, documents);
+        bulkPrintService.printRespondentDocuments(caseDetails, authorisationToken, documents);
+    }
+
+    private void updateGeneralApplicationLatestDocument(CaseDetails caseDetails, CaseDocument document) {
+        Map<String, Object> caseData = caseDetails.getData();
+        caseData.put(GENERAL_APPLICATION_DOCUMENT_LATEST, document);
+        caseData.put(GENERAL_APPLICATION_DOCUMENT_LATEST_DATE, LocalDate.now());
+    }
+
+    private void resetStateToGeneralApplicationPrestate(CaseDetails caseDetails) {
+        Map<String, Object> caseData = caseDetails.getData();
+        String generalApplicationPreState = (String) caseData.get(GENERAL_APPLICATION_PRE_STATE);
+        if (generalApplicationPreState != null) {
+            caseData.put(STATE, generalApplicationPreState);
+        }
+    }
+
+    private CaseDocument prepareGeneralApplicationDirectionsOrderDocument(CaseDetails caseDetails, String authorisationToken) {
+        CaseDetails caseDetailsCopy = documentHelper.deepCopy(caseDetails, CaseDetails.class);
+        Map<String, Object> caseData = caseDetailsCopy.getData();
+
+        addContestedCourtDetails(caseDetailsCopy);
```

Comment:
there are shared methods that will add these values for you, check out `prepareLetterTemplateData` method in `DocumentHelper`

Reply:
They also set other fields not required on this template so decided to KISS and leave it like that.

Reply:
in that case, split the prepareLetterTemplateData method into 2 methods with 1 having only the fields you require and call it directly

Reply:
I think adding properties to the case data that don't get used in the letter yet is going to be less hassle going forward than having to update multiple places if we need to change it. My ultimate goal is for all documents to call the same method to populate a set of case data that is re-used on MOST documents. also, don't use hardcoded strings.